integration-tests/cli/resource-deleter.it.js

```diff
@@ -0,0 +1,176 @@
+import { createRequestBuilder } from '@commercetools/api-request-builder'
+import { createHttpMiddleware } from '@commercetools/sdk-middleware-http'
+import ResourceDeleter from '@commercetools/resource-deleter'
+import { getCredentials } from '@commercetools/get-credentials'
+import { exec } from 'mz/child_process'
+import { version } from '@commercetools/resource-deleter/package.json'
+import {
+  carts,
+  categories,
+  channels,
+  customerGroups,
+  customers,
+  customObjects,
+  inventoryEntries,
+  payments,
+  productDiscounts,
+  products,
+  productTypes,
+  reviews,
+  shippingMethods,
+  taxCategories,
+  types,
+  zones,
+} from './helpers/resource-delete.data'
+import { createData, clearData } from './helpers/utils'
+
+let projectKey
+if (process.env.CI === 'true') projectKey = 'resource-deleter-int-test'
+else projectKey = process.env.npm_config_projectkey
+
+describe('Resource Deleter', () => {
+  let apiConfig
+  let resourceDeleter
+  let logger: {
+    error: () => {},
+    warn: () => {},
+    info: () => {},
+    debug: () => {},
+  }
+  const bin = './integration-tests/node_modules/.bin/resource-deleter'
+
+  function getResource(resource) {
+    const client = createClient({
+      middlewares: [
+        createAuthMiddlewareForClientCredentialsFlow({ ...apiConfig, fetch }),
+        createHttpMiddleware({ host: apiConfig.apiUrl, fetch }),
+      ],
+    })
+
+    const service = createRequestBuilder({
+      projectKey: apiConfig.projectKey,
+    })[resource]
+
+    const request = {
+      uri: service.build(),
+      method: 'GET',
+    }
+
+    return client.execute(request)
+  }
+
+  beforeAll(async () => {
+    // Get test credentials
+    const credentials = await getCredentials(projectKey)
+    apiConfig = {
+      host: 'https://auth.sphere.io',
+      apiUrl: 'https://api.sphere.io',
+      projectKey,
+      credentials,
+    }
+
+    // Create resources on API
+    await Promise.all([
+      createData(apiConfig, 'categories', categories),
+      createData(apiConfig, 'channels', channels),
+      createData(apiConfig, 'customerGroups', customerGroups),
+      createData(apiConfig, 'customers', customers),
+      createData(apiConfig, 'customObjects', customObjects),
+      createData(apiConfig, 'inventory', inventoryEntries),
+      createData(apiConfig, 'payments', payments),
+      createData(apiConfig, 'productDiscounts', productDiscounts),
+      createData(apiConfig, 'productTypes', productTypes),
+      createData(apiConfig, 'reviews', reviews),
+      createData(apiConfig, 'taxCategories', taxCategories),
+      createData(apiConfig, 'types', types),
+      createData(apiConfig, 'zones', zones),
+    ])
+    await Promise.all([
+      createData(apiConfig, 'carts', carts),
+      createData(apiConfig, 'products', products),
+      createData(apiConfig, 'shippingMethods', shippingMethods),
+    ])
+  }, 30000)
+
+  // clear resources on API
+  afterAll(async () => {
+    await Promise.all([
```

Comment:
as described above - this can be simplified by iterating over a list of resources or by using `Promise.map`:
```js
await Promise.map(['carts', 'products', 'shippingMethods'], (resource) => clearData(apiConfig, resource))
```